docs(logging): add logging with timestamp enabled (#2992)
nestjs/docs.nestjs.com: content/techniques/logger.md

This should be suitable for most cases. But if you need more customization (like adding and calling custom methods), move to the next section.

#### Logs with timestamps

> To enable timestamp in your application like the Nest system logging we can use the optional options `timestamp: true` when creating the logger.

Reviewer: ```suggestion
You can enable Nest like system logging by providing the optional `timestamp: true` setting when creating the logger instance.
```

```typescript
import { Logger, Injectable } from '@nestjs/common';

@Injectable()
class MyService {
  private readonly logger = new Logger(MyService.name, { timestamp: true }); // optional timestamp parameter to enable timestamps

  doSomething() {
    this.logger.log('Doing something with timestamp here ->');
  }
}
```

> This will make an output like this:

Reviewer: ```suggestion
This will print the output in the following manner:
```

```bash

> [Nest] 19096   - 12/08/2019, 7:12:59 AM   [NestFactory] Doing something with timestamp here -> +5ms

Reviewer: ```suggestion
[Nest] 19096   - 04/19/2024, 7:12:59 AM   [MyService] Doing something with timestamp here +5ms
```

```

> For each logger the timedifference to the last message will be calculated and printed at the end of the logline.

Reviewer 2: ```suggestion
For each logger the time difference to the last message will be calculated and printed at the end of the log line.
```

> For each logger the time difference to the last message will be calculated and printed at the end of the log line.

Reviewer: ```suggestion
Notice the `+5ms` at the end of the line above. 

For each log statement, the time difference from the last message will be calculated and printed at the end of the line.
```
